.github/workflows/nodejs.yml

```diff
@@ -0,0 +1,39 @@
+name: Node.js
+
+on:
+  pull_request:
+    branches: [master]
+  push:
+    branches: [master]
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        node-version: [12, 14, 16, 18]
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm install --legacy-peer-deps
```

Comment:
Is it worth it to create a `package-lock.json` and use `npm ci` here to ensure reproducible builds?

Reply:
Eventually, yes. I'm hoping to add a separate PR adding a `package-lock.json` file, once at least this and #463 have been merged.